### PR TITLE
OOP school library: preserve data

### PR DESCRIPTION
**Pull request by Ibtisam34**
- [ ]  In this project I had  preserved the data from my app and load preserved data on startup.

### CODE REVIEW

File: main.rb [hunk 1]
```diff
@@ -1,4 +1,4 @@
-require './book'
+# require './data'
```
Comment (youmari) [inline main.rb]: - [ ] it would be better to remove the commented code in order to keep the file clean and pro

File: library.rb [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  puts "Name: #{person[:name]}"
+
+  puts "Age: #{person[:age]}"
+
+  puts "Parent Permission: #{person[:parent_permission]}"
+
+  puts "Rentals: #{person[:rentals].size}"
+
+  puts
+end
```
Comment (youmari) [inline library.rb]: - [ ] When I run the app and add some books and students and teachers and exit and re-run the app it doesn't preserve data as expected it should display all the books and people that were created previously so please make sure to fix that in order to complete the requirements 
- and you should also handle the case when a `(file).json` is missing or empty it should create a file then

**Review by youmari — CHANGES_REQUESTED** [review]
Hi @Ibtisam34 @AhmadiGu , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
✔️ Great commit messages
✔️ PRO PR
✔️ No linter error 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: library.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+people = if File.exist?('people.json') && !File.zero?('people.json')
+           JSON.parse(File.read('people.json'), symbolize_names: true)
+         else
+           []
+         end
+rentals = if File.exist?('rentals.json') && !File.zero?('rentals.json')
+            JSON.parse(File.read('rentals.json'), symbolize_names: true)
+          else
+            []
+          end
```
Comment (DeliceLydia) [inline library.rb:17]: - [ ] Changes requested by the previous reviewer were not addressed still when I exit the application data are not saved in the `JSON` files please address that to be ready to move forward

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi Team, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.


 ## Highlights
- The console app is working as expected


 ## Required Changes ♻️
_Check the comments under the review._



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by m05e5 — APPROVED** [review]
**Hi Team**
You did great. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
CleverScornfulBuffalo-size_restricted

## Highlights

✔️ No linters error
✔️  Well done implementing the requested changes 

## Optional suggestions



_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_**m05e5**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._**

**Pull request merged**